basculechecks/basculechecks.go

```diff
@@ -17,46 +19,177 @@ var (
 	ErrNoValidCapabilityFound = errors.New("no valid capability for endpoint")
 )
 
-func CreateValidCapabilityCheck(prefix string, acceptAllMethod string) (func(context.Context, []interface{}) error, error) {
+const (
+	CapabilityKey = "capabilities"
+	ResourceKey   = "allowedResources"
+	PartnerKey    = "allowedPartners"
+)
+
+type capabilityCheck struct {
+	prefixToMatch   *regexp.Regexp
+	acceptAllMethod string
+	measures        *AuthCapabilityCheckMeasures
+}
+
+var defaultLogger = log.NewNopLogger()
+
+func (c *capabilityCheck) Check(ctx context.Context, token bascule.Token) error {
+	// getting full auth
+	auth, ok := bascule.FromContext(ctx)
+	if !ok {
+		c.measures.CapabilityCheckOutcome.With(ClientIDLabel, "", PartnerIDLabel, "", OutcomeLabel, RejectedTokenMissing).Add(1)
+		return ErrNoAuth
+	}
+
+	// getting metric information
+	client := token.Principal()
+	partnerIDs, err := getPartnerIDList(token.Attributes())
+	if err != nil {
+		c.measures.CapabilityCheckOutcome.With(ClientIDLabel, client, PartnerIDLabel, "", OutcomeLabel, RejectedUndeterminedPartnerID).Add(1)
```

Comment:
You could add a helper function is you think it'd make things more concise. Something like this?
```
c.addOneForCapabilityOutcome(ClientIDLabel, "", PartnerIDLabel, "", OutcomeLabel, RejectedTokenMissing)
```
```
func (c *capabilityCheck) addOneForCapabilityOutcome(labels...){
    c.measures.CapabilityCheckOutcome.With(labels ...).Add(1)
}
```